### PR TITLE
Install plugins in a single transaction in tests

### PR DESCRIPTION
**Pull request by jasontedor**
When building clusters for integration tests, today we install plugins sequentially. We recently introduced the ability to install plugins in a single invocation of the install plugin command. Using this can save substantial time starting up JVMs. This commit changes the build infrastructure to install multiple plugins at once when building clusters for integration tests.

For the docs integration tests in particular, where we install many plugins, this change makes a substantial difference. On my laptop, prior to this change, installing the plugins sequentially took 115 seconds. After this change, it takes 14 seconds.

Relates #50924

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java [hunk 1]
```diff
@@ -428,14 +428,27 @@ public synchronized void start() {
 
         createConfiguration();
 
+        final List<String> pluginsToInstall = new ArrayList<>();
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: Is this intermediary collection really necessary? All it's doing is saving us a single `toString()` call later on.

I mainly say that as it's a bit confusing. Having a separate "plugins to install" to me implies that `this.plugins` is some kind of superset of that, which isn't the case. We always install all configured plugins for the cluster.

**Review by mark-vieira — APPROVED** [review]
Minor comment, otherwise LGTM. This should help tremendously especially when running tests locally.

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (jasontedor) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: Note that we sometimes add `x-pack` to the list. However, we can remove that code in master. Let me do that in a follow-up, and then we can remove the extra collection then, too.

**Review by rjernst — APPROVED** [review]
LGTM

File: buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java (hunk 1 above)
Comment (jasontedor) [reply buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java]: I opened #51441 for this.

**Pull request merged**